x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/queries/SemanticQueryBuilder.java

```diff
@@ -187,13 +202,36 @@ private QueryBuilder doRewriteBuildSemanticQuery(SearchExecutionContext searchEx
         if (fieldType == null) {
             return new MatchNoneQueryBuilder();
         } else if (fieldType instanceof SemanticTextFieldMapper.SemanticTextFieldType semanticTextFieldType) {
-            if (inferenceResults == null) {
+            if (embeddingsProvider == null) {
                 // This should never happen, but throw on it in case it ever does
                 throw new IllegalStateException(
                     "No inference results set for [" + semanticTextFieldType.typeName() + "] field [" + fieldName + "]"
                 );
             }
 
+            ModelRegistry modelRegistry = modelRegistrySupplier.get();
+            if (modelRegistry == null) {
+                throw new IllegalStateException("Model registry has not been set");
```

Comment:
We may need to test this, to make sure it actually should always return a 500/trigger a serverless alert, similar to some other alerts we've been seeing for semantic queries.